contrib/semgrep.yml

```diff
@@ -0,0 +1,10 @@
+rules:
+
+- id: inline-from-impls
+  languages:
+    - rust
+  message: Use inline attribute on `From` implementations.
+  patterns:
+    - pattern-regex: 'From<[A-Za-z]+> for [A-Za-z]+ \{$\n'
```

Comment:
**suggestion:** You don't have to use regex. semgrep supports Rust, but has some limitations around macros handling. This looks like something that should be possible to handle with normal  (non-regex) syntatic matchers.

Reply:
@tcharding https://semgrep.dev/playground/s/AdWQ

Reply:
Oh cool, that is way more powerful!

Reply:
I had a play and couldn't work out how to match on attributes (because the `#` is a comment in yaml). Will come back to it. Mentioning unless you know the solution off the top of your head.

Reply:
Oh. Yeah. Last time I tried the macros were the tricky one.

Reply:
So attribute matching works https://semgrep.dev/playground/s/0z1r , but seems like to in `impl` blocks. :D

I've found some examples in https://github.com/returntocorp/semgrep/blob/develop/tests/patterns/rust/attribute_matching.sgrep

Reply:
I feel a bit disgusted by it, but that's the best I was able to get to work: https://semgrep.dev/playground/s/qDGR

Using `focus-metavariable`, `#[$ARGS]`, `$...ARGS`, etc. should be possible to do it better in theory, but each seems to not work with Rust attributes for reasons unknown.

Reply:
ok, thanks man. Will return to this.